src/averageAges.js

```diff
@@ -55,7 +80,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => {
+    return onlyWithSon
+      ? child.sex === 'm' && people.some(mother => mother.name === child.mother)
+      : people.some(mother => mother.name === child.mother);
+  });
```

Comment:
you can remove this unnecessary return and rewrite it in a way that is more readable

```
const children = people.filter(child => 
     onlyWithSon
      ? child.sex === 'm' && people.some(mother => mother.name === child.mother)
      : people.some(mother => mother.name === child.mother);
 );
```